### ISSUE TITLE
How to get the detect model on my own datasets?

### ISSUE THREAD

**Issue by ztfmars** [JosephKJ/OWOD#55]
Thanks for authors and creative coders for OWOD, and I have some questions:

- 1 can this code be used for new datasets and get my own detect models for indoors targets? Will it detect all the objects in the room labelled with certain names and uknown （will majority of objects that never appeared in training be labelled "unknow"）? 
I just wonder what will be the "unknow" due to numerous new objects and complex enviroment.  because the "unknow" comes from the training feature lantern space, **I really want to know what will be labelled as "unknow" and what will have no labels in the new datasets?**

- 2 if I want to have trained the model with my own datasets. should I choose to train your Task 1-4 first ? or I just only need to  change the dataset path and use your task1 process?
**can anyone give me a simple way to achieve the goal aimed at training some indoors objections** (and labelling the never appeared objects as "unknow" in general), just as easy as yolo series tutorial for green hands ?

- 3 from the google drive, I only see 3 models and no pth model for task 4. while **is it a must to train the tons of data (all the VOC & COCO) so that i can get the final pth model in task 4?**  where can I get the final pth model after Task 1-4 or the final pre-trained model ?

- 4 I test the model with a indoors video using the following cmd:

`python demo.py --config-file ../configs/OWOD/t1/t1_test.yaml --video-input /home/xxxx/shanghai.mp4 --opt MODEL.WEIGHTS /home/xxxx/t1/model_final.pth`

**using the  t1 model tested in a video  there are many "unknow" objects while using t3 model there are almost no "unknow" objects.
is it a capability degradation of detecting unknow objects?** or something wrong?

Look forward to your help and anwsers, thanks again in advance~~~

**Comment by ztfmars** [comment]
@JosephKJ @salman-h-khan 
look forward to your reply, thx very much!

**Comment by JosephKJ** [comment]
Hi Zhang, 

Thank you for your interest in our work. Please find my responses inline:


> 1 can this code be used for new datasets and get my own detect models for indoors targets? Will it detect all the objects in the room labelled with certain names and uknown （will majority of objects that never appeared in training be labelled "unknow"）? 
I just wonder what will be the "unknow" due to numerous new objects and complex enviroment. because the "unknow" comes from the training feature lantern space, I really want to know what will be labelled as "unknow" and what will have no labels in the new datasets?

"this code be used for new dataset"
Definitely yes, you might have to train OWOD on your new dataset. Please do note that we do require a some validation data to learn the unknown distribution. 

"Will it detect all the objects in the room labelled with certain names and uknown": 
These models were trained on pure academic datasets like MS COCO and VOC. Please note that the images in these datasets are very much curated and would be completely different from an indoor scenes (that you intend to test on).

"I really want to know what will be labelled as "unknow""
Those region proposals that have higher probability of belonging to the unknown Weibull distribution will be classified as an unknown. Please refer to Sec 4.3 in the paper. 

> 2 if I want to have trained the model with my own datasets. should I choose to train your Task 1-4 first ? or I just only need to change the dataset path and use your task1 process?
can anyone give me a simple way to achieve the goal aimed at training some indoors objections (and labelling the never appeared objects as "unknow" in general), just as easy as yolo series tutorial for green hands ?

" should I choose to train your Task 1-4 first": No, you can just use the Task 1 process. 

"a simple way to achieve the goal": I dont have the bandwidth to write a Quick Start Guide on Training with a Custom Dataset. But still, we can customize the methodology for your use-case if required. Please write to me at cs17m18p100001@iith.ac.in if you think that is helpful. 

> 3 from the google drive, I only see 3 models and no pth model for task 4. while is it a must to train the tons of data (all the VOC & COCO) so that i can get the final pth model in task 4? where can I get the final pth model after Task 1-4 or the final pre-trained model ?

It is available in t4_ft/model_final.pth inside owod_backup folder. 

> 4 I test the model with a indoors video using the following cmd: <skipped>
using the t1 model tested in a video there are many "unknow" objects while using t3 model there are almost no "unknow" objects.
is it a capability degradation of detecting unknow objects? or something wrong?

This is interesting, would be interested to see your results. Again, we cannot expect it to work out-of-the-box as your dataset is very different. Always use the model from 't*_ft'.


I hope this clarifies many of your questions, please do reopen this issue if required. Thank you.
